### PR TITLE
test(COREL): add core tests

### PR DESCRIPTION
**Pull request by RitaDias**
### Description

- Add createWrapper in core (with resource options)
- Add bundleForm test
- Add BundleIconEditorPicker test
- Add createBundleDialog test
- Add BundleMenu test
- Replace createWrapper in the releases tool with the new one.

Left some components out since they were either:
- Too simple / UI focused
- Will eventually change

### What to review
- Make sure that the changes to the releases plugin makes sense

A lot of these were taken with inspiration from the tests that Jordan has been writing, so thank you 🙏 🥺

### CODE REVIEW

File: packages/sanity/src/core/releases/tool/BundleDetail.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import {ArrowLeftIcon, PublishIcon, StarIcon} from '@sanity/icons'
```
Comment (pedrobonamin) [inline packages/sanity/src/core/releases/tool/BundleDetail.tsx:1]: This file was changed for `ReleaseDetail` https://github.com/sanity-io/sanity/blob/corel/packages/sanity/src/core/releases/tool/detail/ReleaseDetail.tsx

We should remove this and update the tests

File: packages/sanity/src/core/store/bundles/createBundlesStore.ts [hunk 2]
```diff
@@ -171,7 +172,7 @@ export function createBundlesStore(context: {client: SanityClient | null}): Bund
         // After successful reconnection, fetch the list. Note that the first event is skipped, so
         // this will not occur upon initial connection.
         if (event.type === 'welcome') {
-          return listFetch$
+          return listFetch$.pipe(delay(1_000))
```
Comment (pedrobonamin) [inline packages/sanity/src/core/store/bundles/createBundlesStore.ts:175]: Curious on why this change is needed

Comment (RitaDias) [reply packages/sanity/src/core/store/bundles/createBundlesStore.ts:175]: This looks like part of a rebase that has gone wrong 🫠 
hold please...

**Review by pedrobonamin — APPROVED** [review]
Thanks for this @RitaDias
